### PR TITLE
Feature/17 user list and edit user

### PR DESCRIPTION
**Pull request by shivansh-pandeyy**

### CODE REVIEW

File: src/components/CardComponent/CardComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                id="basic-button"
+                aria-label="settings"
+                onClick={handleClick}
+              >
+                <MoreVertIcon />
+              </IconButton>
+              <Menu anchorEl={anchorEl} onClose={handleClose} open={open}>
+                {menuWithAction &&
+                  menuWithAction.length > 0 &&
+                  menuWithAction.map((menuItem) => (
```
Comment (anand-velotio) [inline src/components/CardComponent/CardComponent.tsx]:
```suggestion
                {!!menuWithAction?.length &&
                  menuWithAction?.map((menuItem) => (
```

File: src/redux/reducers/user.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      });
+    case ActionDefTypes.GET_USERS_LIST_END:
+      return produce(state, (draftState) => {
+        draftState.isProcessing = false;
+        draftState.info = payload;
+      });
+    default:
+      return state;
+  }
+};
```
Comment (anand-velotio) [inline src/redux/reducers/user.ts]: const userReducer = (state = INITIAL_STATE, action: ActionDef) => 
  produce(state, (draftState) => {
    const { type, payload } = action;
    switch (type) {
      case ActionDefTypes.GET_USERS_LIST_START:
        draftState.isProcessing = true;
        break;
      case ActionDefTypes.GET_USERS_LIST_END:
        draftState.isProcessing = false;
        draftState.info = payload;
         break;
       default:
         break;
   }
};

Comment (anand-velotio) [reply src/redux/reducers/user.ts]: produce can be used at the top

File: src/redux/reducers/user.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  type: ActionDefTypes.GET_USERS_LIST_END;
+  payload: UserListEndObj[];
+}
+
+type ActionDef = ActionUserListStart | ActionUserListEnd;
+
+const userReducer = (state = INITIAL_STATE, action: ActionDef) => {
+  const { type, payload } = action;
+  switch (type) {
+    case ActionDefTypes.GET_USERS_LIST_START:
```
Comment (anand-velotio) [inline src/redux/reducers/user.ts]: Rename to `IS_USERS_LIST_LOADING`

File: src/redux/reducers/user.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      return produce(state, (draftState) => {
+        draftState.isProcessing = true;
+      });
+    case ActionDefTypes.GET_USERS_LIST_END:
```
Comment (anand-velotio) [inline src/redux/reducers/user.ts]: Rename to `GET_USERS_LIST`

**Review by anand-velotio — CHANGES_REQUESTED** [review]

File: src/components/CardComponent/CardComponent.tsx (hunk 1 above)
Comment (shivansh-pandeyy) [reply src/components/CardComponent/CardComponent.tsx]: done

File: src/redux/reducers/user.ts (hunk 2 above)
Comment (shivansh-pandeyy) [reply src/redux/reducers/user.ts]: done

File: src/redux/reducers/user.ts (hunk 3 above)
Comment (shivansh-pandeyy) [reply src/redux/reducers/user.ts]: done

File: src/redux/reducers/user.ts (hunk 4 above)
Comment (shivansh-pandeyy) [reply src/redux/reducers/user.ts]: done

**Review by anand-velotio — APPROVED** [review]

**Pull request merged**